src/Controls/src/Core/Layout/GridLayout.cs

```diff
@@ -1,26 +1,67 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 using Microsoft.Maui.Layouts;
 
 // This is a temporary namespace until we rename everything and move the legacy layouts
 namespace Microsoft.Maui.Controls.Layout2
 {
 	public class GridLayout : Layout, IGridLayout
 	{
-		readonly List<IGridRowDefinition> _rowDefinitions = new();
-		readonly List<IGridColumnDefinition> _columnDefinitions = new();
+		[TypeConverter(typeof(ColumnDefinitionCollectionTypeConverter))]
+		public ColumnDefinitionCollection ColumnDefinitions
+		{
+			get; set;
+		} = new ColumnDefinitionCollection { new ColumnDefinition { Width = GridLength.Auto } };
+
+		[TypeConverter(typeof(RowDefinitionCollectionTypeConverter))]
+		public RowDefinitionCollection RowDefinitions
+		{
+			get; set;
+		} = new RowDefinitionCollection { new RowDefinition { Height = GridLength.Auto } };
+
+		public static readonly BindableProperty RowProperty = BindableProperty.CreateAttached("Row", 
+			typeof(int), typeof(GridLayout), default(int), validateValue: (bindable, value) => (int)value >= 0);
+
+		public static readonly BindableProperty RowSpanProperty = BindableProperty.CreateAttached("RowSpan", 
+			typeof(int), typeof(GridLayout), 1, validateValue: (bindable, value) => (int)value >= 1);
+
+		public static readonly BindableProperty ColumnProperty = BindableProperty.CreateAttached("Column", 
+			typeof(int), typeof(GridLayout), default(int), validateValue: (bindable, value) => (int)value >= 0);
+
+		public static readonly BindableProperty ColumnSpanProperty = BindableProperty.CreateAttached("ColumnSpan", 
+			typeof(int), typeof(GridLayout), 1, validateValue: (bindable, value) => (int)value >= 1);
 
-		public IReadOnlyList<IGridRowDefinition> RowDefinitions => _rowDefinitions;
-		public IReadOnlyList<IGridColumnDefinition> ColumnDefinitions => _columnDefinitions;
+		public static readonly BindableProperty RowSpacingProperty = BindableProperty.Create("RowSpacing", typeof(double), 
+			typeof(GridLayout), 0d, propertyChanged: (bindable, oldValue, newValue) => ((GridLayout)bindable).InvalidateMeasure());
 
-		public double RowSpacing { get; set; }
-		public double ColumnSpacing { get; set; }
+		public static readonly BindableProperty ColumnSpacingProperty = BindableProperty.Create("ColumnSpacing", typeof(double), 
+			typeof(GridLayout), 0d, propertyChanged: (bindable, oldValue, newValue) => ((GridLayout)bindable).InvalidateMeasure());
 
+		IReadOnlyList<IGridRowDefinition> IGridLayout.RowDefinitions => RowDefinitions.ToList().AsReadOnly();
+		IReadOnlyList<IGridColumnDefinition> IGridLayout.ColumnDefinitions => ColumnDefinitions.ToList().AsReadOnly();
```

Comment:
Is this making a copy? Maybe this should be wrapped into a `new ReadOnlyList(xxx)` instead to avoid copies on each layout? Or maybe the RowDefinitionCollection should also just be a readonly list? and return that exactly?

Reply:
I'll see what I can do.

Reply:
Updated those to use ReadOnlyCastingList.